task/istio-canary-release/0.1/samples/application/app-gateway.yaml

```diff
@@ -0,0 +1,48 @@
+apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
+metadata:
+  name: app-gateway
+spec:
+  selector:
+    istio: ingressgateway
+  servers:
+    - port:
+        number: 80
+        name: http
+        protocol: HTTP
+      hosts:
+        - "*"
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: DestinationRule
+metadata:
+  name: myapp
+spec:
+  host: myapp
+  subsets:
+    - name: v1
+      labels:
+        version: v1
+    - name: v2
+      labels:
+        version: v2
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
```

Comment:
You have already created Virtual Service in Task so do we need this again?

Reply:
This is just a demo `VirtualService`. A user can provide the `VirtualService` along with other istio manifests via the `workspace` or in case if they just want to change the `VirtualService` then they can do it using the task without mounting any `workspace`. Please let me know your thoughts on this